tests/upgrade/test_upgrade.py

```
@@ -0,0 +1,108 @@
+import time
+import sys
+import os
+import threading
```

Ruff

`threading` imported but unused

Remove unused import: `threading`

(F401)